src/serverroutes.js

```diff
@@ -629,4 +633,65 @@ module.exports = function(app, saveSecurityConfig, getSecurityConfig) {
       })
     )
   })
+
+  app.post('/restore', (req, res, next) => {
+    const busboy = new Busboy({ headers: req.headers })
+    busboy.on('file', (fieldname, file, filename, encoding, mimetype) => {
+      try {
+        if (!filename.endsWith('.backup')) {
+          res.status(400).send('Invalid backup file')
+          return
+        }
+        const unzipStream = unzipper.Extract({ path: app.config.configPath })
```

Comment:
I think we should sanitize the input somehow, I feel quezy about just unzipping something.
- upper limit to file size
- only known files in the settings base directory?
- i guess plugins settings and data directories can not be effectively sanitized, there can be dragons there

Should we report to the user something user friendly about what the backup contained? 

Should there be a way to see into the backup file without restoring, like a preview of what there is?

Reply:
All sounds good